library/core/src/ptr/non_null.rs

```diff
@@ -1795,12 +1832,75 @@ mod verify {
         }
     }
 
-    // pub const unsafe fn new(ptr: *mut T) -> Option<Self>
+    // pub const fn new(ptr: *mut T) -> Option<Self>
     #[kani::proof_for_contract(NonNull::new)]
     pub fn non_null_check_new() {
         let mut x: i32 = kani::any();
         let xptr = &mut x;
         let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+
+    // pub const unsafe fn add(self, count: usize) -> Self
+    #[kani::proof_for_contract(NonNull::add)]
+    pub fn non_null_check_add() {
+        const SIZE: usize = 100000;
+        // Randomize pointer offset within array bound
+        let offset = kani::any_where(|x| * x <= SIZE as isize);
+        kani::assume(offset >= 0);
+        // Create a non-deterministic array of size SIZE
+        let arr: [i8; SIZE] = kani::any();  
+        // Get a raw pointer to the array
+        let raw_ptr: *mut i8 = arr.as_ptr() as *mut i8;  
+        // NonNull pointer to the random offset
+        let ptr = unsafe { NonNull::new(raw_ptr.offset(offset)).unwrap()};
+        // Create a non-deterministic count value
+        let count: usize = kani::any();  
+        
+        unsafe {
+            // Add a positive offset to pointer
+            let result = ptr.add(count);
+        }
+    }
+
+    // pub fn addr(self) -> NonZero<usize>
+    #[kani::proof_for_contract(NonNull::addr)]
+    pub fn non_null_check_addr() {  
+        // Create NonNull pointer & get pointer address
+        let mut x: i32 = kani::any();
+        let nonnull_xptr = NonNull::new(&mut x as *mut i32).unwrap();
+        let address = nonnull_xptr.addr();
+    }
+
+    // pub fn align_offset(self, align: usize) -> usize
+    #[kani::proof_for_contract(NonNull::align_offset)]
+    pub fn non_null_check_align_offset() {
+        // Create NonNull pointer
+        let mut x: i8 = kani::any();
+        let nonnull_xptr = NonNull::new(&mut x as *mut i8).unwrap();
+    
+        // Call align_offset with valid align value
+        let align: usize = kani::any();
+        kani::assume(align.is_power_of_two());
+        nonnull_xptr.align_offset(align);
+    }
+
+    // pub fn align_offset(self, align: usize) -> usize
+    #[kani::should_panic]
+    #[kani::proof_for_contract(NonNull::align_offset)]
+    pub fn non_null_check_align_offset_negative() {
+        // Non-deterministic input array of i8 (signed 8-bit integers)
```

Comment:
The comment incorrectly refers to an array.

Reply:
Thanks for catching!